packages/client/src/devices/InputMediaDeviceManager.ts

```diff
@@ -63,16 +63,23 @@ export abstract class InputMediaDeviceManager<
    * Starts stream.
    */
   async enable() {
-    if (this.state.status === 'enabled') return;
-    this.enablePromise = this.unmuteStream();
-    try {
-      await this.enablePromise;
-      this.state.setStatus('enabled');
-      this.enablePromise = undefined;
-    } catch (error) {
-      this.enablePromise = undefined;
-      throw error;
-    }
+    if (this.state.pendingStatus === 'enabled') return;
+    const signal = this.nextAbortableStatusChangeRequest('enabled');
```

Comment:
2. Cancel any pending request that hasn't started yet.
3. Set pending status to `enabled`.